### PR TITLE
feat(kit): support empty value in `InputCount`

### LINKED ISSUE #3281: UIKIT-6218

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #3281

### CODE REVIEW

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import {WAIT_BEFORE_SCREENSHOT} from '@demo-integrations/support/properties/shared.entities';
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts:1]:
```
Module not found: Error:
Can't resolve '@demo-integrations/support/properties/shared.entities' in
'/home/runner/work/taiga-ui/taiga-ui/projects/demo-integrations/cypress/tests/desktop/kit/input-count'
```

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts (hunk 1 above)
Comment (splincode) [reply projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts]: fixed

File: projects/kit/components/input-count/input-count.component.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ import {
 import {NgControl} from '@angular/forms';
 import {
     AbstractTuiControl,
```
Comment (waterplea) [inline projects/kit/components/input-count/input-count.component.ts:15]: Is this still used?

File: projects/kit/components/input-count/input-count.component.ts [hunk 3]
```diff
@@ -296,12 +301,18 @@ export class TuiInputCountComponent
...
+            : formatNumber(
+                  value,
+                  null,
+                  this.numberFormat.decimalSeparator,
+                  this.numberFormat.thousandSeparator,
+              );
+    }
+
+    private isNotNumber(value: number | null): value is null {
+        return Number.isNaN(value as any) || !isPresent(value);
```
Comment (waterplea) [inline projects/kit/components/input-count/input-count.component.ts:316]: If you change order, you won't need to typecast

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/input-count/input-count.component.ts (hunk 2 above)
Comment (splincode) [reply projects/kit/components/input-count/input-count.component.ts:15]: yes, in v3.x
image

in v2.x
image

**Review by nsbarsukov — APPROVED** [review]

File: projects/kit/components/input-count/input-count.component.ts (hunk 3 above)
Comment (splincode) [reply projects/kit/components/input-count/input-count.component.ts]: fixed

**Pull request merged**